lib/srv/db/redis/engine.go

```diff
@@ -282,8 +283,9 @@ func (e *Engine) createOnClientConnectFunc(ctx context.Context, sessionCtx *comm
 	// See: https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/auth-iam.html#auth-iam-limits
 	// So we must check that the database supports IAM auth and that the
 	// ElastiCache user has IAM auth enabled.
+	// Same applies for AWS MemoryDB.
 	case e.isAWSIAMAuthSupported(ctx, sessionCtx):
-		credFetchFn := elasticacheIAMTokenFetchFunc(sessionCtx, e.Auth)
+		credFetchFn := awsIAMTokenFetchFunc(sessionCtx, e.Auth)
```

Comment:
nit: Not sure if this can happen but since credFetchFn can be nil it should be resonable to check agains credFetchFn before passing the nil args to `fetchCredentialsOnConnect` to prevent panic.

Reply:
Good catch! Added an error in func return and then propagated that error up. will do some manual testing before merge.